File: scripts/update_ostaninth_data.py
#!/usr/bin/env python3
"""
Update ostaninth test data to use Twitter ID 187893504
"""

import pandas as pd
import os

NEW_TWITTER_ID = "187893504"
NEW_HANDLE = "ostaninth"
DATA_DIR = os.path.join(os.getcwd(), "data_sample", "ostaninth")

def update_parquet_file(filename, description):
    filepath = os.path.join(DATA_DIR, filename)

    if not os.path.exists(filepath):
        print(f"⚠️  {filename} not found, skipping...")
        return

    print(f"\n📄 Updating {filename}...")

    try:
        # Read parquet file
        df = pd.read_parquet(filepath)
        print(f"  Found {len(df)} rows")

        # Update account_id column if it exists (snake_case)
        if 'account_id' in df.columns:
            old_ids = df['account_id'].unique()
            print(f"  Old account_id values: {old_ids[:3]}...")  # Show first 3
            df['account_id'] = NEW_TWITTER_ID
            print(f"  ✓ Updated account_id column to {NEW_TWITTER_ID}")
        elif 'accountId' in df.columns:  # Try camelCase too
            old_ids = df['accountId'].unique()
            print(f"  Old accountId values: {old_ids[:3]}...")
            df['accountId'] = NEW_TWITTER_ID
            print(f"  ✓ Updated accountId column to {NEW_TWITTER_ID}")

        # Update username column if it exists
        if 'username' in df.columns:
            old_names = df['username'].unique()
            print(f"  Old username values: {old_names[:3]}...")
            df['username'] = NEW_HANDLE
            print(f"  ✓ Updated username column to {NEW_HANDLE}")
        elif 'userName' in df.columns:  # Try camelCase too
            df['userName'] = NEW_HANDLE
            print(f"  ✓ Updated userName column to {NEW_HANDLE}")

        # Write back to parquet
        df.to_parquet(filepath, index=False)
        print(f"  ✅ Saved updated {filename}")

    except Exception as e:
        print(f"  ❌ Error updating {filename}: {e}")

def main():
    print("🔧 Updating ostaninth data for Twitter ID:", NEW_TWITTER_ID)
    print("📁 Data directory:", DATA_DIR)

    # Check if pandas and pyarrow are installed
    try:
        import pyarrow
        print("✓ pandas and pyarrow are installed\n")
    except ImportError:
        print("❌ Error: pyarrow is required. Install with:")
        print("   pip install pandas pyarrow")
        return

    # Update parquet files
    update_parquet_file("tweets_df.parquet", "Main tweets data")
    update_parquet_file("clustered_tweets_df.parquet", "Clustered tweets data")

    print("\n✅ All updates complete!")
    print("\nNext steps:")
    print("1. Upload the entire 'data_sample/ostaninth' folder to your S3 bucket")
    print("2. Make sure it's at: s3://your-bucket/ostaninth/")
    print("3. Log in with Twitter account @ostaninth (ID: 187893504)")
    print("4. Click 'Get the Analysis' to see your data!")

if __name__ == "__main__":
    main()

File: scripts/update-ostaninth-data.ts
#!/usr/bin/env node

/**
 * Update ostaninth test data to use Twitter ID 187893504
 * Note: Run with Node.js, not Bun, due to parquet-wasm compatibility
 */

const { readFileSync, writeFileSync } = require("fs");
const { join } = require("path");

const NEW_TWITTER_ID = "187893504";
const NEW_HANDLE = "ostaninth";
const DATA_DIR = join(process.cwd(), "data_sample", "ostaninth");

async function updateParquetFiles() {
  console.log("Loading parquet libraries...");
  const parquet = await import("parquet-wasm/node/arrow2");
  await parquet.default();

  const arrow = await import("apache-arrow");

  // Update tweets_df.parquet
  const tweetsPath = join(DATA_DIR, "tweets_df.parquet");
  console.log("\n📄 Updating tweets_df.parquet...");

  try {
    const tweetsBuffer = readFileSync(tweetsPath);
    const tweetsTable = parquet.readParquet(new Uint8Array(tweetsBuffer));
    const tweetsArrowTable = arrow.tableFromIPC(tweetsTable.intoIPCStream());

    console.log(`  Found ${tweetsArrowTable.numRows} rows`);

    // Get the schema and data
    const schema = tweetsArrowTable.schema;
    const data: any = {};

    // Copy all columns
    for (const field of schema.fields) {
      const columnData = tweetsArrowTable.getChild(field.name);
      if (field.name === 'accountId') {
        // Replace all accountId values with new Twitter ID
        const newAccountIds = Array(tweetsArrowTable.numRows).fill(NEW_TWITTER_ID);
        data[field.name] = newAccountIds;
        console.log(`  ✓ Updated accountId column to ${NEW_TWITTER_ID}`);
      } else if (field.name === 'userName') {
        // Replace userName values with new handle
        const newUserNames = Array(tweetsArrowTable.numRows).fill(NEW_HANDLE);
        data[field.name] = newUserNames;
        console.log(`  ✓ Updated userName column to ${NEW_HANDLE}`);
      } else {
        // Keep original data
        data[field.name] = columnData?.toArray();
      }
    }

    // Create new Arrow table
    const newTable = arrow.tableFromArrays(data);

    // Write to parquet
    const newParquetBuffer = parquet.writeParquet(newTable);
    writeFileSync(tweetsPath, Buffer.from(newParquetBuffer));
    console.log(`  ✅ Saved updated tweets_df.parquet`);

  } catch (error) {
    console.error(`  ❌ Error updating tweets_df.parquet:`, error);
  }

  // Update clustered_tweets_df.parquet
  const clusteredPath = join(DATA_DIR, "clustered_tweets_df.parquet");
  console.log("\n📄 Updating clustered_tweets_df.parquet...");

  try {
    const clusteredBuffer = readFileSync(clusteredPath);
    const clusteredTable = parquet.readParquet(new Uint8Array(clusteredBuffer));
    const clusteredArrowTable = arrow.tableFromIPC(clusteredTable.intoIPCStream());

    console.log(`  Found ${clusteredArrowTable.numRows} rows`);

    const schema = clusteredArrowTable.schema;
    const data: any = {};

    for (const field of schema.fields) {
      const columnData = clusteredArrowTable.getChild(field.name);
      if (field.name === 'accountId') {
        const newAccountIds = Array(clusteredArrowTable.numRows).fill(NEW_TWITTER_ID);
        data[field.name] = newAccountIds;
        console.log(`  ✓ Updated accountId column to ${NEW_TWITTER_ID}`);
      } else if (field.name === 'userName') {
        const newUserNames = Array(clusteredArrowTable.numRows).fill(NEW_HANDLE);
        data[field.name] = newUserNames;
        console.log(`  ✓ Updated userName column to ${NEW_HANDLE}`);
      } else {
        data[field.name] = columnData?.toArray();
      }
    }

    const newTable = arrow.tableFromArrays(data);
    const newParquetBuffer = parquet.writeParquet(newTable);
    writeFileSync(clusteredPath, Buffer.from(newParquetBuffer));
    console.log(`  ✅ Saved updated clustered_tweets_df.parquet`);

  } catch (error) {
    console.error(`  ❌ Error updating clustered_tweets_df.parquet:`, error);
  }
}

async function main() {
  console.log("🔧 Updating ostaninth data for Twitter ID:", NEW_TWITTER_ID);
  console.log("📁 Data directory:", DATA_DIR);

  await updateParquetFiles();

  console.log("\n✅ All updates complete!");
  console.log("\nNext steps:");
  console.log("1. Upload the entire 'data_sample/ostaninth' folder to your S3 bucket");
  console.log("2. Make sure it's in the root of the bucket (not in a subfolder)");
  console.log("3. Log in with Twitter account @ostaninth (ID: 187893504)");
  console.log("4. Click 'Get the Analysis' to see your data!");
}

main().catch(console.error);
